SimPEG/utils/io_utils/io_utils_pf.py

```diff
@@ -296,7 +296,7 @@ def read_gg3d_ubc(obs_file):
                     d.append(factor * temp[3 : 3 + n_comp])
                     wd.append(temp[3 + n_comp :])
                 ii += 1
-            except:
+            except IOError:
```

Comment:
I think there's a couple possible errors that should be caught here, all related to the above lines.
- Should error on file read error
- Should error if the array is unable to be converted to floats,

Reply:
You're right. I think the main issue is that it's running a lot of things within a single `try`, shadowing possible errors. I unpacked some portions of this code. The `readline()` is not likely to file, but to return an empty string if it hits the EOF. Now the `try` only contains the parsing of the line to the array and I improved the error messages to be more descriptive. I also fixed the line number shown in the messages: line numbers usually start in 1 and we are already reading the first two lines, so the actual line number at each iteration is `ii + 3`.